AppInspector/MetaDataHelper.cs

```diff
@@ -11,11 +13,22 @@ namespace Microsoft.ApplicationInspector.Commands
 {
     /// <summary>
     /// Provides utilty help specific to aggregating metadata from analyze cmd matches while isolating scanned data from that process
+    /// Hides complexity i.e. threaded scanning so that caller gets simple list<T> that is presorted and consistent each scan
     /// </summary>
     public class MetaDataHelper
     {
-        //inhouse common properties to capture
-        private readonly Dictionary<string, Regex> _propertyTagSearchPatterns;
+        public ConcurrentDictionary<string, byte> PackageTypes { get; set; }
+        public ConcurrentDictionary<string, byte> AppTypes { get; set; }
+        public ConcurrentDictionary<string, byte> FileNames { get; set; }
+        public ConcurrentDictionary<string, byte> UniqueTags { get; set; }
+        public ConcurrentDictionary<string, byte> UniqueDependencies { get; set; }
+        public ConcurrentDictionary<string, byte> Outputs { get; set; }
+        public ConcurrentDictionary<string, byte> Targets { get; set; }
+        public ConcurrentDictionary<string, byte> FileExtensions { get; set; }
+        public ConcurrentDictionary<string, byte> CPUTargets { get; set; }
+        public ConcurrentDictionary<string, byte> CloudTargets { get; set; }        
+        public ConcurrentDictionary<string, byte> OSTargets { get; set; }
+        ConcurrentDictionary<string,MetricTagCounter> TagCounters { get; set; }
```

Comment:
why is this one not public but the others are?

Are these supposed to be public in the first place?  Shouldn't they be internal since this is designed to hide complexity from users?

Reply:
Some are visible from AnalyzeCommand, others aren't and are processed when a match is added.  I'll recheck each.

Reply:
Analyze command is in the same namespace so it shouldn't need to be public to access it.

Public means people writing code using the app inspector library can access these properties (and they should be documented and supported)

Do you intend to publicly expose the dictionaries? I thought not.

Reply:
internal works...thanks